lib/Froxlor/Validate/Validate.php

```diff
@@ -62,32 +62,77 @@ public static function validate($str, $fieldname, $pattern = '', $lng = '', $emp
 		\Froxlor\UI\Response::standard_error($lng, $fieldname, $throw_exception);
 	}
 
-	/**
-	 * Checks whether it is a valid ip
-	 *
-	 * @param string $ip
-	 *        	ip-address to check
-	 * @param bool $return_bool
-	 *        	whether to return bool or call \Froxlor\UI\Response::standard_error()
-	 * @param string $lng
-	 *        	index for error-message (if $return_bool is false)
-	 * @param bool $allow_localhost
-	 *        	whether to allow 127.0.0.1
-	 * @param bool $allow_priv
-	 *        	whether to allow private network addresses
-	 * @param bool $allow_cidr
-	 *        	whether to allow CIDR values e.g. 10.10.10.10/16
-	 *        	
-	 * @return string|bool ip address on success, false on failure
-	 */
-	public static function validate_ip2($ip, $return_bool = false, $lng = 'invalidip', $allow_localhost = false, $allow_priv = false, $allow_cidr = false, $throw_exception = false)
+    /**
+     * Converts CIDR to a netmask address
+     *
+     * @thx to https://stackoverflow.com/a/5711080/3020926
+     * @param string $cidr
+     *
+     * @return string
+     */
+    public static function cidr2NetmaskAddr ($cidr) {
+
+        $ta = substr ($cidr, strpos ($cidr, '/') + 1) * 1;
+        $netmask = str_split (str_pad (str_pad ('', $ta, '1'), 32, '0'), 8);
+
+        foreach ($netmask as &$element) {
+            $element = bindec ($element);
+        }
+
+        return implode ('.', $netmask);
+    }
+
+    /**
+     * Checks if an $address (IP) is IPv6
+     *
+     * @param $address
+     *
+     * @return bool
+     * @thx https://stackoverflow.com/a/13677565/3020926
+     */
+    public static function is_ipv6($address) {
```

Comment:
Don't understand why this should be needed, there's `filter_var()` for checking whether an IP address is ipv6 or not

Reply:
I'll leave that because there is something called mapped ipv6 address. Like ::ffff:10.20.30.40 which translates to a ipv4 address but in ipv6 "language". If I understand that correctly, tho. That function takes care of it, while `filter_var` apparanetly does not (Did not test!).

Reply:
Yeah I've read that too...I would drop that for now because if you support that here you'd have to support this notation everywhere

Reply:
`filter_var($address, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6)` does return false when `$address` is `2620:0:2d0:200::7/32.

Reply:
still don't get the point why you need that - you have the `$ip_cidr` array at the point where you call `self::is_ipv6()` (iirc) which for my understanding should be:
```php
$ip_cidr = [
    0 => 2620:0:2d0:200::7,
    1 => 32
];
```
So, just use `filter_var` on `$ip_cidr[0]`

Reply:
Changed.